tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/join/ControlledWorkerBindJoin.java

```diff
@@ -45,12 +45,7 @@ protected TaskCreator determineTaskCreator(TupleExpr expr, BindingSet bs) {
 		final TaskCreator taskCreator;
 		if (expr instanceof StatementTupleExpr) {
 			StatementTupleExpr stmt = (StatementTupleExpr) expr;
-			if (stmt.hasFreeVarsFor(bs)) {
-				taskCreator = new BoundJoinTaskCreator(strategy, stmt);
-			} else {
-				expr = new CheckStatementPattern(stmt, queryInfo);
-				taskCreator = new CheckJoinTaskCreator(strategy, (CheckStatementPattern) expr);
-			}
+			taskCreator = new BoundJoinTaskCreator(strategy, stmt);
```

Comment:
this is the relevant functional change: the VALUES based implementation can also be used for bind joins without free variables without functional change.

Note that in very early versions (where the bind join was implemented with a UNION construct) there was a need to distinguish the cases. This is a left over